src/shared/firewall-util-private.h

```diff
@@ -5,18 +5,45 @@
 #include <stdint.h>
 
 #include "in-addr-util.h"
+#include "sd-netlink.h"
 
 enum fw_backend {
         FW_BACKEND_NONE,
 #if HAVE_LIBIPTC
         FW_BACKEND_IPTABLES,
 #endif
+        FW_BACKEND_NFTABLES,
 };
 
 struct fw_ctx {
         enum fw_backend fw_backend;
+        union {
+                struct {
+                        sd_netlink *nfnl;
+                } nft;
+        } u;
```

Comment:
i'd not bother with the subunion and substruct... It#s not that a million backends are going to pop-up any time soon which want to overload this...